### PR TITLE
Fix a bug with the definition of `max_shufflable_size`

### PR DESCRIPTION
**Pull request by kspieks**
As pointed out in #163, many of our extrapolative samplers do not properly utilized the `random_state` when shuffling the indices. The issue comes from the previous definition of `max_shufflable_size=min(n_test_samples, n_val_samples)`. If `val_size=0`, such as when calling `train_test_split()`, the `max_shufflable_size` is the min of 0 (from `val_size`) and some positive number (from `test_size`). Since `max_shufflable_size` gets defined as 0 in this case, nothing is shuffled inside the `get_sorted_cluster_counter()` method, which is causing the behavior seen in #163 i.e., the random seed effectively isn't used. 

This commit fixes this issue by reassigning `max_shufflable_size` to be the min of train and test or val and test, depending on which is appropriate. I welcome feedback on alternative methods to address this. Looking forward to your thoughts!

### CODE REVIEW

**Review by JacksonBurns — COMMENTED** [review]
Thanks for the quick patch @kspieks! I like this approach and it seems well thought out.

I'll tack on a commit with a new unit test to ensure that when calling `train_test_split` with an extrapolative sampler the results can actually be shuffled differently by altering `random_state`.

**Comment by kspieks** [comment]
Thanks for adding the unit test and reviewing!
